Analysis of Some RNAseq Study
=============================

Background and Sample Preparation
---------------------------------


Sequence Alignment and Transcript Assembly
------------------------------------------
Bowtie2 was used to align these samples to the **ENSEMBL_M38** reference genome with a gtf file from release **69**.  

Transcript Analysis
-------------------
To analyse differential expression at the gene level, DESeq and Cufflinks were used.  

Gene Counts Table
#################

First databases of exons and transcripts are generated from the Ensembl genome assembly.  The current biomart release of ensembl is M38.69

```{r gene-counts-setup, echo=TRUE}
require(GenomicFeatures)
require(biomaRt)
working_directory = "/data3/davebrid/RNAseq/"
setwd(working_directory)
#make a database of transcripts from the ensembl assembly
#first get the current release 69 from ensembl
txdb <- makeTranscriptDbFromBiomart(biomart="ensembl",dataset = 'mmusculus_gene_ensembl')
#make exon and transcript annotation objects
save(txdb, file="txdb.Robject")
exons <- exons(txdb, columns=c('gene_id', 'tx_id', 'tx_name', 'tx_chrom', 'tx_strand', 'exon_id', 'exon_name', 'cds_id', 'cds_name', 'cds_chrom', 'cds_strand', 'exon_rank'))
save(exons, file="exons.Robject")
transcripts <- transcripts(txdb, columns=c('gene_id', 'tx_id', 'tx_name', 'exon_id', 'exon_name', 'exon_chrom', 'exon_strand', 'cds_id', 'cds_name', 'cds_chrom', 'cds_strand', 'exon_rank'))
save(transcripts, file="transcripts.Robject")
```

Next, each accepted_hits.bam file is read, and overlaps are counted relative to these exon and transcript databases.

```{r transcript-counts, echo=TRUE}
require(GenomicRanges)
require(Rsamtools)

#set list of sample ids as a vector
sample_ids = seq(12849,12858)

transcript.countsTable <- data.frame(
  row.names = as.vector(unlist(elementMetadata(transcripts)['tx_name'])))
exon.countsTable <- data.frame(
  row.names = as.vector(unlist(elementMetadata(exons)['exon_name'])))

#this forloop iterates over the sample_ids and generates exon and transcript counts for each sample_id
for(sample_id in sample_ids) {
  #read alignment
  align <- readBamGappedAlignments(sprintf("tophat_out/Sample_%s/accepted_hits.bam", sample_id))
  #count the overlapping reads for the transcripts
  transcript.counts <- countOverlaps(transcripts, align)
  #reassign to a specific transcript.counts object.
  assign(sprintf("transcript.counts.%s", sample_id), transcript.counts)
  #add this column to the countsTable
  transcript.countsTable <- cbind(transcript.countsTable,transcript.counts)
  remove(transcript.counts)
  #count the overlapping reads for the exons
  exon.counts <- countOverlaps(exons, align)
  #reassign to a specific transcript.counts object.
  assign(sprintf("exon.counts.%s", sample_id), exon.counts)
  #add this column to the countsTable
  exon.countsTable <- cbind(exon.countsTable, exon.counts)  
  #remove the align, transcript.counts and exon.counts objects for the next loop
  remove(align)
  remove(exon.counts)
  }

summary(transcript.countsTable)
summary(exon.countsTable)

#write these two counts tables to csv files.
write.csv(transcript.countsTable, "transcript_counts_table.csv")
write.csv(exon.countsTable, "exon_counts_table.csv")
```

Session Information
-------------------

For the R session, the package versions were:
```{r}
sessionInfo()
```
